arrows/core/transform_detected_object_set.cxx

```diff
@@ -0,0 +1,276 @@
+
+  config->set_value( "src_camera_krtd_file_name", src_camera_krtd_file_name,
+                     "Source camera KRTD file name path" );
+
+  config->set_value( "dest_camera_krtd_file_name", dest_camera_krtd_file_name,
+                     "Destination camera KRTD file name path" );
+
+  return config;
+}
+
+
+// ---------------------------------------------------------------------------
+void
+transform_detected_object_set::
+set_configuration( vital::config_block_sptr config_in )
+{
+  vital::config_block_sptr config = this->get_configuration();
+
+  config->merge_config( config_in );
+  this->src_camera_krtd_file_name =
+    config->get_value< std::string > ( "src_camera_krtd_file_name" );
+  this->dest_camera_krtd_file_name =
+    config->get_value< std::string > ( "dest_camera_krtd_file_name" );
+
+  this->src_camera =
+    kwiver::vital::read_krtd_file( this->src_camera_krtd_file_name );
+  this->dest_camera =
+    kwiver::vital::read_krtd_file( this->dest_camera_krtd_file_name );
+}
+
+
+// ---------------------------------------------------------------------------
+bool
+transform_detected_object_set::
+check_configuration( vital::config_block_sptr config ) const
+{
+  kwiver::vital::config_difference cd( this->get_configuration(), config );
+  const auto key_list = cd.extra_keys();
+
+  if ( ! key_list.empty() )
+  {
+    LOG_WARN( logger(), "Additional parameters found in config block that are "
+                        "not required or desired: "
+                        << kwiver::vital::join( key_list, ", " ) );
+  }
+
+  return true;
+}
+
+
+// ---------------------------------------------------------------------------
+vector_3d
+transform_detected_object_set::
+backproject_to_ground( const kwiver::vital::camera_perspective_sptr camera,
+                       const vector_2d img_pt ) const
+{
+  auto ground_plane = vector_4d(0, 0, 1, 0);
+  return this->backproject_to_plane( camera, img_pt, ground_plane );
+}
+
+// ---------------------------------------------------------------------------
+vector_3d
+transform_detected_object_set::
+backproject_to_plane( const kwiver::vital::camera_perspective_sptr camera,
+                      const vector_2d img_pt,
+                      const vector_4d plane ) const
+{
+  // Back an image point to a specified world plane
+  vector_2d npt_ = camera->intrinsics()->unmap(img_pt);
+  auto npt = vector_3d(npt_(0), npt_(1), 1.0);
+
+  matrix_3x3d M = camera->rotation().matrix().transpose();
+
+  auto n = vector_3d(plane(0), plane(1), plane(2));
+  double d = plane(3);
+
+  vector_3d Mt = M * camera->translation();
+  vector_3d Mp = M * npt;
+
+  return Mp * (n.dot( Mt ) - d) / n.dot( Mp ) - Mt;
+}
+
+// ---------------------------------------------------------------------------
+Eigen::Matrix<double, 8, 3>
+transform_detected_object_set::
+backproject_bbox( const kwiver::vital::camera_perspective_sptr camera,
+                  const vector_4d box ) const
```

Comment:
It might make the code easier to understand if you just pass `box` as a `vital::bounding_box<double>` all the way into this function.  Then you can access `box.min_x()` instead of `box(0)` and it's more clear what that is.  There is nothing technically wrong with what you have here, but using the more specific type will help self-document the function a little better.

This change is not a requirement, just a suggestion.

Reply:
Agreed that that's a more transparent approach, should be fixed now.